prime-router/src/test/kotlin/ReportTests.kt

```diff
@@ -909,128 +933,288 @@ class ReportTests {
     }
 
     @Test
-    fun `test generateReportAndUploadBlob for hl7`() {
+    fun `test generateReportAndUploadBlob for fhir`() {
```

Comment:
Added 3 integration tests here that connect to a fake blobstore in order to verify the name generation of generateReportAndUploadBlob is correct, among other stuff.